### ISSUE TITLE
series.data is empty after delay

### ISSUE THREAD

**Issue by janbols** [amcharts/amcharts4#2452]
**Bug description**

Getting the data of a series is empty, even after setting it previously.

When constructing a new series without setting its own data immediately, but instead, set the `series.data` field after a delay, the `series.data` field becomes empty sometimes.

Instead, when setting the `series.data` immediately during the creation of the series, the problem does not occur.

There's a codepen example available on https://codepen.io/janbols/pen/oNjEpZJ?editors=0010
This is an adapted version found on https://www.amcharts.com/docs/v4/tutorials/dynamically-adding-and-removing-series/#Adding_series but with the following changes:
* each series uses its own data set
* `series.data` is only set after a delay
* it logs the `series.data` for each added series

After adding a few series, you can notice in the console log that some series don't contain any data. 

**Environment (if applicable)**
AMCharts 4.9.19
Firefox 75.0

**Comment by zeroin** [comment]
Hi,

Could you explain - do you mean that the first time console.log's empty array for the second series like in the picture?
![Annotation 2020-05-07 193520](https://user-images.githubusercontent.com/5336255/81320574-16cb3180-909a-11ea-8f50-49ae40801401.jpg)

**Comment by janbols** [comment]
Yes indeed, though it's clearer to notice when you add more series. They
will remain empty, even when setting its series.data field. After waiting a
few millis, the array will be empty again.
Hope I made myself clear.

Best regards
Jan

**Comment by zeroin** [comment]
Could you modify the script so that the problem would be visible?

As now I don't see it. You call addSeries twice, each time with data set in one second. When the timeout method is called first time, you set data for the first series and do console.log. So it's normal that the second series does not yet have data, as the second timeout not happened yet. When the second timeout happens, you set data for the second series and now both series has data.

**Comment by janbols** [comment]
Ok, I adapted the script:
Every second 1 series is added. The data is set without delay; but within a setTimeout block. 
After 10 seconds, no more series are added and the data field for each series is shown in console log.

The output of my run is shown below:
```
series data for 0 
Array []
series data for 1 
Array []
series data for 2 
Array(19) [ {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, … ]
series data for 3 
Array(19) [ {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, … ]
series data for 4 
Array []
series data for 5 
Array(19) [ {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, … ]
series data for 6 
Array []
series data for 7 
Array(19) [ {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, … ]
series data for 8 
Array(19) [ {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, … ]
```
​
You can see that some series have data, some don't. Modifying the script by moving the assignment of the data field outside the `setTimeout` block results in all fields having data.

Hope this helps.

**Comment by zeroin** [comment]
Thanks - I managed to reproduce and fix it in dev. We will soon release an update and will let you know.

**Comment by martynasma** [comment]
Fixed in [4.9.20](https://github.com/amcharts/amcharts4/releases/tag/4.9.20).

## [4.9.20] - 2020-05-11

### Changed
- `hit` and `up` events on nested objects will now respect correct bubbling up order.
- If `rtl` is not set directly on the `Label`, it will now take `rtl` value not from its parent but from `baseSprite` (performance enhancement).
- `<desc>` element with amCharts attribution removed from generated SVG.

### Fixed
- Export: PDF export will now fallback to regular fonts if bold fonts don't exist.
- Export: Advanced PDF exports with extra content were broken since last release.
- Export: Image export was essentially broken in IEs.
- `options.nonce` was not working properly on EDGE.
- Event `selectedned` was not dispatched by `Cursor` if released outside the chart div and behavior was set to `select*`.
- If all series were removed and then added quickly to the chart with a legend it could result a JS error.
- Setting `data` directly on a series not right after it was created could result `series.data` array to be shown as empty.
- It was impossible to set `SwitchButton` as active initially.
- Fixed snapping cursor to series when data was missing: if series had gaps in data and `cursor.snapToSeries` was set, the snapping was not working properly.
- When adding data to `XYSeries` which was `XYCharScrollbar`'s series, the raw data was incorrectly modified.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
